eng/scripts/generate_from_source_pom.py

```diff
@@ -176,6 +176,9 @@ def add_project_to_dependency_and_module_mappings(file_path: str, project_depend
 
     for dependency in dependencies:
         dependency_identifier = create_artifact_identifier(dependency)
+        if dependency_identifier is None:
```

Comment:
This is added to support surefire plugin `dependenciesToScan` config: it allows to run tests from other projects
```xml
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>3.0.0-M3</version> <!-- {x-version-update;org.apache.maven.plugins:maven-surefire-plugin;external_dependency} -->
        <configuration>
          <dependenciesToScan>
            <dependency>com.azure:azure-core</dependency>
          </dependenciesToScan>
        </configuration>
      </plugin>
```

It this case, `<dependency>` does not represent maven dependency and irrelevant to this script

Reply:
This should also allow for libraries to use BOMs for versioning, not sure if that is the intended side-effect

Reply:
`dependenciesToScan` also supports the same artifact identifier as `maven-enforcer-plugin`, so the use of `dependenciesToScan` could be updated to use that identifier format which works with the versioning tooling.

Reply:
this script is not used for versioning, it seems it builds a temp pom for testing or validation. It just failed on `<dependency>` without `<artifactId>` or `<groupId>`. I guess it never runs for boms and nothing else uses `dependenciesToScan` so far in our repo.